src/pages/List.js

```diff
@@ -26,14 +49,27 @@ const List = props => {
             return <ErrorMessage />;
           }
 
+          if (!isLoading) {
+            setShoppingList(data);
+          }
+
           return isLoading ? (
             // TODO: Make a display list function is listContext.js
             <Loading />
           ) : (
-            <ul>
-              {setShoppingList(data)}
+            <ul className="shopping-list">
               {shoppingList.map((item, index) => (
-                <li key={index}>{item.name}</li>
+                <li key={index}>
+                  <label>
+                    <input
+                      type="checkbox"
+                      //checked is a reflection of a field on the item. it shouldn’t be local state. you should be able to have something like checked={isChecked(item.lastDatePurchased)} .
+                      checked={isChecked(item.lastDatePurchased)}
```

Comment:
Nice job!

Reply:
@stevelikesmusic gave great advice and i figured it out the `item.lastDatepurchased` prior to looking at his code. Thank you!